modules/ghlint-extensions/src/test/kotlin/net/twisterrob/ghlint/model/EffectiveScopesKtTest.kt

```diff
@@ -0,0 +1,210 @@
+package net.twisterrob.ghlint.model
+
+import io.kotest.matchers.collections.beEmpty
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+import net.twisterrob.ghlint.testing.load
+import org.intellij.lang.annotations.Language
+import org.junit.jupiter.api.Test
+
+class EffectiveScopesKtTest {
+
+	private fun loadPermissions(@Language("yaml") yaml: String): Permissions {
+		val file = load(yaml)
+		return (file.content as Workflow).permissions!!
+	}
+
+	@Test fun `single read permission`() {
+		val permissions = loadPermissions(
+			"""
+				on: push
+				permissions:
+				  contents: read
+				jobs:
+				  test:
+				    uses: reusable/workflow.yml
+			""".trimIndent(),
+		)
+
+		permissions.effectiveScopes shouldBe setOf(
+			Scope(Permission.CONTENTS, Access.READ),
+		)
+	}
+
+	@Test fun `single write permission includes read`() {
+		val permissions = loadPermissions(
+			"""
+				on: push
+				permissions:
+				  pull-requests: write
+				jobs:
+				  test:
+				    uses: reusable/workflow.yml
+			""".trimIndent(),
+		)
+
+		permissions.effectiveScopes shouldBe setOf(
+			Scope(Permission.PULL_REQUESTS, Access.READ),
+			Scope(Permission.PULL_REQUESTS, Access.WRITE),
+		)
+	}
+
+	@Test fun `all none includes nothing`() {
+		val permissions = loadPermissions(
+			"""
+				on: push
+				permissions:
+				  actions: none
+				  attestations: none
+				  checks: none
+				  contents: none
+				  deployments: none
+				  id-token: none
+				  issues: none
+				  #metadata: none
```

Comment:
So it seems it's not possible to declare this explicitly: 
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idpermissions

Reply:
Removed them, and added `discussions`.

ca6ef3cd325394a47acf345d07bb492a413ba513